ansible/roles/couchdb/tasks/main.yml

```diff
@@ -80,16 +81,16 @@
     url: "{{ couchdb_base_url }}/{{ localsettings.COUCH_DATABASE_NAME }}"
   register: couch_response
 
-- block:
-    - name: Add CouchDB database
-      uri:
-        url: "{{ couchdb_base_url }}/{{ localsettings.COUCH_DATABASE_NAME }}"
-        method: PUT
-
-    - name: Set CouchDB username and password
-      uri:
-        url: "{{ couchdb_base_url }}/_config/admins/{{ localsettings.COUCH_USERNAME }}"
-        method: PUT
-        body: '"{{ localsettings.COUCH_PASSWORD }}"'
-        body_format: raw
+- name: Add CouchDB database
+  uri:
+    url: "{{ couchdb_base_url }}/{{ localsettings.COUCH_DATABASE_NAME }}"
+    method: PUT
   when: couch_response.status == 404
+
+- name: Set CouchDB username and password
+  uri:
+    url: "{{ couchdb_base_url }}/_config/admins/{{ localsettings.COUCH_USERNAME }}"
+    method: PUT
+    body: '"{{ localsettings.COUCH_PASSWORD }}"'
+    body_format: raw
+  when: update_config|changed
```

Comment:
I think this used to run when couch_response.status == 404. does that still need to happen?